test/es6/proxybugs.js

```diff
@@ -37,6 +37,24 @@ var tests = [
                 assert.areEqual(target.x, 20, "target object should work as expected even after proxy object is added to map");
             });
         }
+    },
+    {
+        name: "Assertion validation : returning descriptor during getOwnPropertyDescriptor should not pollute the descriptor",
+        body() {
+            var target = {};
+            var handler = {};
+            handler['defineProperty'] = function () {
+                assert.fail("This function will not be called as 'getOwnPropertyDescriptor' will return a descriptor");
+            };
+            
+            handler['getOwnPropertyDescriptor'] = function (t, property) {
+                Object.defineProperty(t, 'abc', { set: function () { } });
```

Comment:
Ok - will do that.

---
In reply to: [96047714](https://github.com/Microsoft/ChakraCore/pull/2374#discussion_r96047714) [](ancestors = 96047714,96045829,95927783)